git-tar/function/ops_test.go

```diff
@@ -103,105 +103,79 @@ func Test_getRepositoryURL_whenGetTokenReturnsError_WhenRepositoryIsPrivate(t *t
+			expectedError: false,
 		},
 		{
 			title:         "Templates with single added custom repository",
 			envRepos:      "https://github.com/my-custom/repo.git",
 			expectedRepos: append(formalRepos, "https://github.com/my-custom/repo.git"),
+			expectedError: false,
 		},
 		{
 			title:         "Templates with two added custom repositories without spaces",
 			envRepos:      "https://github.com/my-custom/repo.git,https://github.com/another/repo.git",
 			expectedRepos: append(formalRepos, ([]string{"https://github.com/my-custom/repo.git", "https://github.com/another/repo.git"})...),
+			expectedError: false,
 		},
-	}
-	for _, test := range tests {
-		os.Setenv("custom_templates", test.envRepos)
-		t.Run(test.title, func(t *testing.T) {
-			templateRepos := formatTemplateRepos()
-			for _, templateRepo := range templateRepos {
-				for final, expectedRepo := range test.expectedRepos {
-					if expectedRepo == templateRepo {
-						continue
-					}
-					if final == len(test.expectedRepos) {
-						t.Errorf("Expecting repositories: \n`%s` \ngot: \n`%s`",
-							strings.Join(test.expectedRepos, " "),
-							strings.Join(templateRepos, " "))
-					}
-
-				}
-			}
-		})
-	}
-}
-
-func Test_formatTemplateReposUnvalid(t *testing.T) {
-	formalRepos := []string{
-		"https://github.com/openfaas/templates",
-		"https://github.com/openfaas-incubator/node8-express-template.git",
-		"https://github.com/openfaas-incubator/golang-http-template.git",
-	}
-
-	tests := []struct {
-		title         string
-		envRepos      string
-		expectedRepos []string
-	}{
 		{
 			title:         "Variable set invalid",
 			envRepos:      " ",
 			expectedRepos: formalRepos,
+			expectedError: true,
 		},
 		{
 			title:         "Variable set with random symbols",
 			envRepos:      "123randomzxc",
 			expectedRepos: formalRepos,
+			expectedError: true,
 		},
 		{
 			title:         "Invalid github URLs (Missing `https://`)",
 			envRepos:      "www.github.com/my-custom/repo.git",
 			expectedRepos: formalRepos,
+			expectedError: true,
 		},
 		{
 			title:         "Setting values with spaces between commas",
 			envRepos:      " , https://github.com/my-custom/repo.git, https://github.com/another/repo.git, ",
 			expectedRepos: append(formalRepos, ([]string{"https://github.com/my-custom/repo.git", "https://github.com/another/repo.git"})...),
+			expectedError: true,
 		},
 	}
 	for _, test := range tests {
 		os.Setenv("custom_templates", test.envRepos)
-		t.Run(test.title, func(t *testing.T) {
-			templateRepos := formatTemplateRepos()
-			for _, templateRepo := range templateRepos {
-				for final, expectedRepo := range test.expectedRepos {
-					if expectedRepo == templateRepo {
-						continue
-					}
-					if final == len(test.expectedRepos) {
-						t.Errorf("Expecting repositories: \n`%s` \ngot: \n`%s`",
-							strings.Join(test.expectedRepos, " "),
-							strings.Join(templateRepos, " "))
-					}
-
-				}
+		templateRepos, err := formatTemplateRepos()
```

Comment:
Instead of printing test names in the errors, we can use sub-tests like we have in Derek and some of the other code.